source/dhtslib/vcf/record.d

```diff
@@ -459,8 +711,12 @@ struct VCFRecord
     /// "If flt_id is PASS, all existing filters are removed first. If other than PASS, existing PASS is removed."
     int addFilter(string f)
     {
-        return bcf_add_filter(this.vcfheader.hdr, this.line, 
-            bcf_hdr_id2int(this.vcfheader.hdr, BCF_DT_ID, toStringz(f)));
+        
+        if(f == "") hts_log_warning(__FUNCTION__, "filter was empty");
```

Comment:
Probably, I tried not to alter previous methods too much. This was an older method. I will look into it.

Reply:
Confirmed* returns -1; since `f` is a D string, you could do the following:

```
if (fid == -1) hts_log_warning(__FUNCTION__, format("ignoring filter not found in header: %s", f.empty() ? "(empty)" : f));
```


*-> https://github.com/samtools/htslib/blob/238fe32d8c7aa05d3ac75d2249c61d8e268be58f/vcf.c#L3287-L3292

Reply:
addressed in
0910089